tests/FunctionsTest.php

```diff
@@ -270,6 +270,18 @@ public function testParsesRequestMessagesWithUriWhenHostIsNotFirst()
         $this->assertEquals('http://foo.com/', (string) $request->getUri());
     }
 
+    public function testParsesRequestMessagesWithFullUri()
+    {
+        $req = "GET https://www.google.com:443/search?q=foobar HTTP/1.1\r\nHost: www.google.com\r\n\r\n";
+        $request = Psr7\parse_request($req);
+        $this->assertEquals('GET', $request->getMethod());
+        $this->assertEquals('/search?q=foobar', $request->getRequestTarget());
```

Comment:
Shouldn't request target be set to `https://www.google.com:443/search?q=foobar`? I think it's basically supposed to match the path part of a request's start-line.